Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALCalibExtractor.h

```diff
@@ -88,16 +95,46 @@ class EMCALCalibExtractor
 
   /// \brief For now a dummy function to calibrate the time.
   template <typename... axes>
-  o2::emcal::TimeCalibrationParams calibrateTime(boost::histogram::histogram<axes...>& hist)
+  o2::emcal::TimeCalibrationParams calibrateTime(boost::histogram::histogram<axes...>& hist, double minTime = 0, double maxTime = 1000)
   {
+
+    auto histReduced = boost::histogram::algorithm::reduce(hist, boost::histogram::algorithm::shrink(minTime, maxTime), boost::histogram::algorithm::shrink(0, NCELLS));
+
     o2::emcal::TimeCalibrationParams TCP;
-    TCP.addTimeCalibParam(1234, 600, 0);
+
+#ifdef WITH_OPENMP
+    if (mNThreads < 1) {
+      mNThreads = std::min(omp_get_max_threads(), NCELLS);
```

Comment:
17644 threads is probably out of order for local machines :) Needs to be checked with @shahor02 and @chiarazampolli where to put the maximum here, I guess omp threads will need to be shared among components.

Reply:
Yes NCells is unrealistic. But like this it will use all available resources when calling the omp_get_max_threads(), right?